math/matrix/strassenmatrixmultiply.go

```diff
@@ -9,93 +9,219 @@
+		if err != nil {
+			return Matrix[T]{}, err
+		}
+
+		A3, err := A21.Add(A22)
+		if err != nil {
+			return Matrix[T]{}, err
+		}
+
+		A4, err := A11.Add(A12)
+		if err != nil {
+			return Matrix[T]{}, err
+		}
+
+		A5, err := B11.Add(B12)
+		if err != nil {
+			return Matrix[T]{}, err
+		}
+
+		A6, err := B21.Add(B22)
+		if err != nil {
+			return Matrix[T]{}, err
+		}
 		//
-		S1, _ := B12.Subtract(B22)
-		S2, _ := B21.Subtract(B11)
-		S3, _ := A21.Subtract(A11)
-		S4, _ := A12.Subtract(A22)
+		S1, err := B12.Subtract(B22)
+		if err != nil {
+			return Matrix[T]{}, err
+		}
+		S2, err := B21.Subtract(B11)
+		if err != nil {
+			return Matrix[T]{}, err
+		}
+		S3, err := A21.Subtract(A11)
+		if err != nil {
+			return Matrix[T]{}, err
+		}
+		S4, err := A12.Subtract(A22)
+		if err != nil {
+			return Matrix[T]{}, err
+		}
 		// Recursive steps
-		M1 := A1.StrassenMatrixMultiply(A2)
-		M2 := A3.StrassenMatrixMultiply(B11)
-		M3 := A11.StrassenMatrixMultiply(S1)
-		M4 := A22.StrassenMatrixMultiply(S2)
-		M5 := A4.StrassenMatrixMultiply(B22)
-		M6 := S3.StrassenMatrixMultiply(A5)
-		M7 := S4.StrassenMatrixMultiply(A6)
-		//
-		A7, _ := M1.Add(M4)
-		A8, _ := A7.Add(M7)
-		A9, _ := M1.Add(M3)
-		A10, _ := A9.Add(M6)
+		M1, err := A1.StrassenMatrixMultiply(A2)
+		if err != nil {
+			return Matrix[T]{}, err
+		}
+		M2, err := A3.StrassenMatrixMultiply(B11)
+		if err != nil {
+			return Matrix[T]{}, err
+		}
+		M3, err := A11.StrassenMatrixMultiply(S1)
+		if err != nil {
+			return Matrix[T]{}, err
+		}
+		M4, err := A22.StrassenMatrixMultiply(S2)
+		if err != nil {
+			return Matrix[T]{}, err
+		}
+		M5, err := A4.StrassenMatrixMultiply(B22)
+		if err != nil {
+			return Matrix[T]{}, err
+		}
+		M6, err := S3.StrassenMatrixMultiply(A5)
+		if err != nil {
+			return Matrix[T]{}, err
+		}
+		M7, err := S4.StrassenMatrixMultiply(A6)
+
+		if err != nil {
+			return Matrix[T]{}, err
+		} //
+		A7, err := M1.Add(M4)
+
+		if err != nil {
+			return Matrix[T]{}, err
+		}
+		A8, err := A7.Add(M7)
+
+		if err != nil {
+			return Matrix[T]{}, err
+		}
+		A9, err := M1.Add(M3)
+
+		if err != nil {
+			return Matrix[T]{}, err
+		}
+		A10, err := A9.Add(M6)
+
+		if err != nil {
+			return Matrix[T]{}, err
+		}
 		// Calculate result submatrices
-		C11, _ := A8.Subtract(M5)
-		C12, _ := M3.Add(M5)
-		C21, _ := M2.Add(M4)
-		C22, _ := A10.Subtract(M2)
+		C11, err := A8.Subtract(M5)
+		if err != nil {
+			return Matrix[T]{}, err
+		}
+		C12, err := M3.Add(M5)
+		if err != nil {
+			return Matrix[T]{}, err
+		}
+		C21, err := M2.Add(M4)
+		if err != nil {
+			return Matrix[T]{}, err
+		}
+		C22, err := A10.Subtract(M2)
+		if err != nil {
+			return Matrix[T]{}, err
+		}
 
 		// Combine subMatrices into the result matrix
 		var zeroVal T
 		C := New(n, n, zeroVal)
 
 		for i := 0; i < mid; i++ {
 			for j := 0; j < mid; j++ {
-				val, _ := C11.Get(i, j)
+				val, err := C11.Get(i, j)
+				if err != nil {
+					panic("copyMatrix.Set error: " + err.Error())
```

Comment:
I don't know how I missed it - you can surely do something better than `panic` here.

Reply:
Yeah..I missed it too. Thanks for correction.🙂